source/slang/slang-emit-wgsl.cpp

```diff
@@ -298,6 +298,31 @@ void WGSLSourceEmitter::emit(const AddressSpace addressSpace)
     }
 }
 
+static const char* getWgslImageFormat(IRTextureTypeBase* type)
+{
+    ImageFormat imageFormat = type->hasFormat() ? (ImageFormat)type->getFormat() : ImageFormat::unknown;
+    switch (imageFormat)
+    {
+    case ImageFormat::rgba8:       return "rgba8unorm";
```

Comment:
No r8unorm/rg8unorm?

Reply:
r16unorm/rg16unorm/rgba16unorm and snorm formats as well.

Reply:
Right. One channel format can only be 32bit.
Two channel format can only be 64bit in total.